packages/webpack/src/plugins/nx-webpack-plugin/lib/apply-web-config.ts

```diff
@@ -332,6 +332,7 @@ export function applyWebConfig(
 
   config.output = {
     ...config.output,
+    assetModuleFilename: '[name].[contenthash][ext]',
```

Comment:
I think this needs to be `hash` at least in development mode for HMR to work. Maybe this needs to be a ternary assetModuleFilename: ('mode === 'production' ? '[name].[contenthash][ext]' : '[name].[hash:8][ext]' )
or we update the `getOutputHashFormat` use that.

> Hash: This is a hash generated every time that webpack compiles, in dev mode this is good for cache busting during development but shouldn't be used for long term caching of your files. This will overwrite the Hash on every build of your project.